engine/access/state_stream/backend/handler_test.go

```diff
@@ -508,6 +511,75 @@ func TestEventStream(t *testing.T) {
 	}
 }
 
+// TestGetRegisterValues tests the register values.
+func TestGetRegisterValues(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	testHeight := uint64(1)
+
+	// load test register IDs + values
+	testIds := flow.RegisterIDs{
+		flow.UUIDRegisterID(0),
+		flow.AccountStatusRegisterID(unittest.AddressFixture()),
+		unittest.RegisterIDFixture(),
+	}
+	testValues := []flow.RegisterValue{
+		[]byte("uno"),
+		[]byte("dos"),
+		[]byte("tres"),
+	}
+	invalidIDs := append(testIds, flow.RegisterID{}) // valid + invalid IDs
+
+	api := ssmock.NewAPI(t)
+	api.On("GetRegisterValues", testIds, testHeight).Return(testValues, nil)
+	// wrong height
+	api.On("GetRegisterValues", testIds, testHeight+1).Return(nil, storage.ErrHeightNotIndexed)
+	// invalid register
+	api.On("GetRegisterValues", invalidIDs, testHeight).Return(nil, storage.ErrNotFound)
+	h := NewHandler(api, flow.Localnet.Chain(), makeConfig(1))
+
+	// invalid message
+	invalidMessage := &executiondata.GetRegisterValuesRequest{
+		RegisterIds: nil,
+	}
+	_, err := h.GetRegisterValues(ctx, invalidMessage)
+	require.Equal(t, status.Code(err), codes.InvalidArgument)
+
+	// valid registers
+	validRegisters := make([]*entities.RegisterID, len(testIds))
+	for i, id := range testIds {
+		validRegisters[i] = convert.RegisterIDToMessage(id)
+	}
+	req := &executiondata.GetRegisterValuesRequest{
+		RegisterIds: validRegisters,
+		BlockHeight: testHeight,
+	}
+	resp, err := h.GetRegisterValues(ctx, req)
+	require.NoError(t, err)
+	require.Equal(t, resp.GetValues(), testValues)
+
+	// unavailable registers (should be atomic)
```

Comment:
what do you mean by `(should be atomic)` here?

Reply:
all-or-nothing, the test is for a list of valid requests + 1 invalid one, the entire request fails instead of returning the valid subset. I'll remove the verbiage as it's not doing its job